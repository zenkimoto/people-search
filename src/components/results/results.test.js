import { render, cleanup, screen } from '@testing-library/react';
import ReactTestRenderer from 'react-test-renderer';
import Results from './results';

afterEach(() => {
  cleanup();
});

it('should not crash if no people are passed into Results component', async () => {
  // Render
  render(<Results />);
});

it('should render a list of people', async () => {
  const people = [
    { id: 1, name: 'John Smith' },
    { id: 2, name: 'Jane Doe' },
  ];

  render(<Results people={people} />);

  // Get list item by test id
  const listItem1 = screen.getByTestId('person-1');
  expect(listItem1).toBeInTheDocument();
  expect(listItem1).toHaveTextContent('John Smith');

  // Get list item by test id
  const listItem2 = screen.getByTestId('person-2');
  expect(listItem2).toBeInTheDocument();
  expect(listItem2).toHaveTextContent('Jane Doe');
});

it('should match snapshot', () => {
  // Example of a snapshot test
  const people = [
    { id: 1, name: 'John Smith' },
    { id: 2, name: 'Jane Doe' },
  ];

  // React Test Renderer library can be found here:
  // https://www.npmjs.com/package/react-test-renderer
  const tree = ReactTestRenderer.create(<Results people={people} />).toJSON();

  expect(tree).toMatchSnapshot();
});
