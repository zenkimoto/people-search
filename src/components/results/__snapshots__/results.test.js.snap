// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="p-10"
>
  <h1
    className="text-lg mb-4"
  >
    Results
  </h1>
  <ul
    className="list-disc"
  >
    <li
      data-testid="person-1"
    >
      John Smith
    </li>
    <li
      data-testid="person-2"
    >
      Jane Doe
    </li>
  </ul>
</div>
`;
